src/pip/_internal/pep425tags.py

```diff
@@ -1,235 +1,37 @@
 """Generate and work with PEP 425 Compatibility Tags."""
 from __future__ import absolute_import
 
-import distutils.util
 import logging
-import platform
 import re
-import sys
-import sysconfig
 
 from pip._vendor.packaging.tags import (
     Tag,
+    compatible_tags,
+    cpython_tags,
+    generic_tags,
     interpreter_name,
     interpreter_version,
     mac_platforms,
 )
-from pip._vendor.six import PY2
 
-import pip._internal.utils.glibc
 from pip._internal.utils.typing import MYPY_CHECK_RUNNING
 
 if MYPY_CHECK_RUNNING:
-    from typing import (
-        Callable, List, Optional, Tuple, Union
-    )
+    from typing import List, Optional, Tuple
+
+    from pip._vendor.packaging.tags import PythonVersion
 
 logger = logging.getLogger(__name__)
 
 _osx_arch_pat = re.compile(r'(.+)_(\d+)_(\d+)_(.+)')
 
 
-def get_config_var(var):
-    # type: (str) -> Optional[str]
-    return sysconfig.get_config_var(var)
-
-
 def version_info_to_nodot(version_info):
```

Comment:
follow up: I suggest moving the call to be changed -- changing `get_supported` to allow for version to be None (and use this value in that case).